### ISSUE TITLE
list.found object

### ISSUE THREAD

**Issue by Graur** [objectionary/eo-collections#27]
Let's implement it, according to [this](https://arxiv.org/pdf/2206.02585.pdf) paper. 
e.g.
```
[] > test
  list > lst
    * 1 2 3
  assert-that > @
    lst.found 2
    $.equal-to 1
```

**Comment by levBagryansky** [comment]
@Graur, I implemented it in pr #25, please close issue.

**Comment by Graur** [comment]
@levBagryansky Thanks!
